### PR TITLE
Gives vending machines a chance of displaying ads

### LINKED ISSUE #16102: Vending machine "product_ads" appear to be completely unused
#### Description of issue

there's a whole buncha `product_ads` defined for every vending machine, but I'm pretty sure they're unused. I've never seen any of the text in the code, and I couldn't find any code that actually *uses* it.
The feature is implemented in other codebases I think, so shouldn't be too hard to get working.

#### Difference between expected and actual behavior

Pull up vending machine menu, see ad
vs
never see ads

#### Steps to reproduce 

- Go to a powered, functional coffee machine
- Open it up and close it ten times, looking for any of the following text:
`"Have a drink!;Drink up!;It's good for you!;Would you like a hot joe?;I'd kill for some coffee!;The best beans in the galaxy.;Only the finest brew for you.;Mmmm. Nothing like a coffee.;I like coffee, don't you?;Coffee helps you work!;Try some tea.;We hope you like the best!;Try our new chocolate!;Admin conspiracies"`
- Buy some items, scouring the window for the same text
- Come to the conclusion that you just wasted a bunch of time because the text isn't ever there

#### Specific information for locating
https://github.com/vgstation-coders/vgstation13/blob/b2b2c2b678afae09c45fe7b770582e2cc9c0bc29/code/game/machinery/vending.dm#L57
https://github.com/vgstation-coders/vgstation13/blob/b2b2c2b678afae09c45fe7b770582e2cc9c0bc29/code/game/machinery/vending.dm#L63

#### Length of time in which bug has been known to occur
I've never seen it working as intended.

#### Server revision
91c316c1c264a59b633cc9c52dd9223146d17621

### PR DESCRIPTION
**Pull request by DamianX**
Closes #16102

Yes, I used `<marquee>` in 2017.

image

:cl:
 * rscadd: Vending machines may display ads when selecting a product. (after 5 years of them being in the code)

### CODE REVIEW

File: code/game/machinery/vending.dm [hunk 1]
```diff
@@ -1161,13 +1182,33 @@ var/global/num_vending_terminals = 1
...
+		"I'd kill for some coffee!",
+		"The best beans in the galaxy.",
+		"Only the finest brew for you.",
+		"Mmmm. Nothing like a coffee.",
+		"I like coffee, don't you?",
+		"Coffee helps you work!",
+		"Try some tea.",
+		"We hope you like the best!",
+		"Try our new chocolate!",
+		"Admin conspiracies"
```
Comment (N3X15) [inline code/game/machinery/vending.dm]: This is horrifically out of place with the other ads for this machine.

File: code/game/machinery/vending.dm [hunk 2]
```diff
@@ -718,8 +715,10 @@ var/global/num_vending_terminals = 1
 		onclose(user, "")
 		return
 
-	var/dat = "<TT><center><b>[vendorname]</b></center><hr /><br>" //display the name, and added a horizontal rule
-	dat += "<b>Select an item: </b><br><br>" //the rest is just general spacing and bolding
+	var/dat = "<TT><center><b>[vendorname]</b></center><hr/>" //display the name, and added a horizontal rule
+	if(prob(ads_chance))
+		dat += "<marquee>[pick(product_ads)]</marquee><hr/>"
```
Comment (N3X15) [inline code/game/machinery/vending.dm]: You forgot `<blink>`, too

**Review by N3X15 — APPROVED** [review]

File: code/game/machinery/vending.dm [hunk 3]
```diff
@@ -1192,8 +1233,23 @@ var/global/num_vending_terminals = 1
...
+	)
+	product_ads = list(
+		"The healthiest!",
+		"Award-winning chocolate bars!",
+		"Mmm! So good!",
+		"Oh my god it's so juicy!",
+		"Have a snack.",
+		"Snacks are good for you!",
+		"Have some more Getmore!",
+		"Best quality snacks straight from mars.",
```
Comment (Kurfursten) [inline code/game/machinery/vending.dm]: Mars

File: code/game/machinery/vending.dm (hunk 1 above)
Comment (SonixApache) [reply code/game/machinery/vending.dm]: 095e84de15d79898a76238f7e94c46df2b096851, although probably before that as well

**Comment by SonixApache** [comment]
Stolen form othercode or written from scratch?
Also should change changelog to `bugfix: vending machine product ads now finally work after 5 years`

**Comment by DamianX** [comment]
@SonixApache I don't know of any othercode that uses this

File: code/game/machinery/vending.dm [hunk 4]
```diff
@@ -2226,8 +2404,16 @@ var/global/num_vending_terminals = 1
 	desc = "A vending machine containing hats."
 	icon_state = "hats"
 	vend_reply = "Take care now!"
-	product_ads = "Buy some hats!;A bare head is absoloutly ASKING for a robusting!"
-	product_slogans = "Warning, not all hats are dog/monkey compatable. Apply forcefully with care.;Apply directly to the forehead.;Who doesn't love spending cash on hats?!;From the people that brought you collectable hat crates, Hatlord!"
+	product_ads = list(
+		"Buy some hats!",
+		"A bare head is absoloutly ASKING for a robusting!"
```
Comment (SonixApache) [inline code/game/machinery/vending.dm]: absolutely*

File: code/game/machinery/vending.dm [hunk 5]
```diff
@@ -2226,8 +2404,16 @@ var/global/num_vending_terminals = 1
...
+	)
+	product_slogans = list(
+		"Warning, not all hats are dog/monkey compatable. Apply forcefully with care.",
```
Comment (SonixApache) [inline code/game/machinery/vending.dm]: compatible*

**Pull request merged**